### PR TITLE
Add `_WILDCARD_TOKEN` special case in `_chop_path`

### LINKED ISSUE #9870: undefer_group with selectin/subqueryload raises exception
### Describe the bug

A query that uses `undefer_group` with either `selectinload` or `subqueryload` raises an `AttributeError` exception.

### Optional link from https://docs.sqlalchemy.org which documents the behavior that is expected

_No response_

### SQLAlchemy Version in Use

2.0.15

### DBAPI (i.e. the database driver)

pysqlite

### Database Vendor and Major Version

SQLite

### Python Version

3.10.11

### Operating system

Linux

### To Reproduce

```python
from sqlalchemy import ForeignKey
from sqlalchemy import create_engine
from sqlalchemy import select
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import Session
from sqlalchemy.orm import mapped_column
from sqlalchemy.orm import relationship
from sqlalchemy.orm import selectinload
from sqlalchemy.orm import undefer_group


class Base(DeclarativeBase):
    pass


class User(Base):
    __tablename__ = 'user'

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(deferred_group='name')

    addresses: Mapped[list['Address']] = relationship()


class Address(Base):
    __tablename__ = 'address'

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey(User.id))
    address: Mapped[str]


engine = create_engine('sqlite://')
Base.metadata.create_all(engine)


with Session(engine) as session:
    user = User(name='name')
    session.add(user)
    session.commit()

with Session(engine) as session:
    session.execute(select(User).options(
        selectinload(User.addresses),
        undefer_group('name'),
    )).scalars().all()
```

### Error

```
Traceback (most recent call last):
  File "/home/phy1729/sqla-demo/demo.py", line 47, in <module>
    )).scalars().all()
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/engine/result.py", line 1784, in all
    return self._allrows()
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/engine/result.py", line 557, in _allrows
    rows = self._fetchall_impl()
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/engine/result.py", line 1691, in _fetchall_impl
    return self._real_result._fetchall_impl()
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/engine/result.py", line 2291, in _fetchall_impl
    return list(self.iterator)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/loading.py", line 218, in chunks
    post_load.invoke(context, path)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/loading.py", line 1515, in invoke
    loader(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/strategies.py", line 3217, in _load_for_path
    self._load_via_parent(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/strategies.py", line 3293, in _load_via_parent
    context.session.execute(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2232, in execute
    return self._execute_internal(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2127, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 293, in orm_execute_statement
    result = conn.execute(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1413, in execute
    return meth(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 483, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1629, in _execute_clauseelement
    compiled_sql, extracted_params, cache_hit = elem._compile_w_cache(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 671, in _compile_w_cache
    compiled_sql = self._compiler(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 288, in _compiler
    return dialect.statement_compiler(dialect, self, **kw)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/compiler.py", line 1426, in __init__
    Compiled.__init__(self, dialect, statement, **kwargs)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/compiler.py", line 867, in __init__
    self.string = self.process(self.statement, **compile_kwargs)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/compiler.py", line 912, in process
    return obj._compiler_dispatch(self, **kwargs)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/visitors.py", line 143, in _compiler_dispatch
    return meth(self, **kw)  # type: ignore  # noqa: E501
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/compiler.py", line 4592, in visit_select
    compile_state = select_stmt._compile_state_factory(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/sql/base.py", line 678, in create_for_statement
    return klass.create_for_statement(statement, compiler, **kw)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 1122, in create_for_statement
    opt.process_compile_state(self)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/strategy_options.py", line 905, in process_compile_state
    self._process(
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/strategy_options.py", line 1377, in _process
    new_path = self._chop_path(start_path, current_path)
  File "/home/phy1729/sqla-demo/venv/lib/python3.10/site-packages/sqlalchemy/orm/strategy_options.py", line 938, in _chop_path
    and c_token != p_token.key  # type: ignore
AttributeError: 'Mapper' object has no attribute 'key'
``` 

### Additional context

The `undefer_group` needs to be at the top level e.g. `selectinload(Foo.bar).undefer_group('baz')` does not error. Issue bisects to 91501e06a17d873902114275d7149ba24973db6a.

I have a potential fix at https://github.com/phy1729/sqlalchemy/commit/b2722932eb7d0ad33300aaf475fc96a11151f5ba, but the contributing docs ask that an issue be opened first.

### PR DESCRIPTION
**Pull request by phy1729**
### Description

Add special case to `_chop_path` for initial `_WILDCARD_TOKEN` matching `_DEFAULT_TOKEN` to allow for a top level `undefer_group` load option with other load options. Fixes #9870.

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [x] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 3f15bb3fbe479d63e9a208ce95a59e22160f692d of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 3f15bb3fbe479d63e9a208ce95a59e22160f692d: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4727

**Comment by zzzeek** [comment]
great thanks for adding a test too

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

needs a changelog file

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4727

File: doc/build/changelog/unreleased_20/9870.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.. change::
+    :tags: bug, orm, regression
+    :tickets: 9870
+
+    Fixed regression in the 2.0 series where a query that used
+    :func:`.undefer_group` with :func:`_orm.selectinload` or
+    :func:`_orm.subqueryload` would raise an ``AttributeError``.
```
Comment (CaselIT) [inline doc/build/changelog/unreleased_20/9870.rst:7]: Feel free to add "Pull request courtesy of Matthew Martin"

Comment (phy1729) [reply doc/build/changelog/unreleased_20/9870.rst]: Added.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 0cf415a7edf6d69ea2ea12b8b10fc8755eefcb8d of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Failed to create a gerrit review, git squash against branch 'main' failed

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 0cf415a7edf6d69ea2ea12b8b10fc8755eefcb8d of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Failed to create a gerrit review, git squash against branch 'main' failed

**Comment by CaselIT** [comment]
Can you rebase or merge the main branch?

**Comment by phy1729** [comment]
Sure; done.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision c073f0904f12909ff660beef6576f6476a725592 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset c073f0904f12909ff660beef6576f6476a725592 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4727

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

thanks very much!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4727

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4727 has been **merged**. Congratulations! :)

**Pull request closed without merging**